### ISSUE TITLE
How can this be used?

### ISSUE THREAD

**Issue by ntrrgc** [jasmine-addons/jasmine_rsvp#1]
I tried adding the Javascript file in the dist folder and it fails due to _`define` being not defined_. I guess this library uses some kind of module management library, maybe RequireJS because there is a `.requirejs` file.

So, what do I need to do in order to use this? Do I have to learn RequireJS, even though I don't plan to use it? Is there any way to use it without that? Would I need to convert my project to RequireJS in order to use this?

**Comment by amireh** [comment]
Yes, the package is built as an AMD module as mentioned in the readme. You will need some kind of AMD loader (like require.js) but it is not necessary to use it for your entire project, just enough to get the dependencies in _before_ you start Jasmine.

How are you running your tests? Do you have them written in a single file?

I can share a snippet of how to manually include require.js, load jasmine_rsvp (and RSVP), and then run the tests.

**Comment by ntrrgc** [comment]
I run my tests using [grunt-contrib-jasmine](https://github.com/gruntjs/grunt-contrib-jasmine), which creates a test runner page with one script tag for each library, source file, and spec file, in the specified order.

**Comment by amireh** [comment]
Then it's easier, check out [grunt-template-jasmine-requirejs](https://github.com/cloudchen/grunt-template-jasmine-requirejs) then do something like this in your grunt task config:

``` javascript
grunt.initConfig({
  connect: {
    test : {
      port : 8000
    }
  },
  jasmine: {
    taskName: {
      src: 'src/**/*.js',
      options: {
        specs: 'spec/*Spec.js',
        helpers: 'spec/*Helper.js',
        host: 'http://127.0.0.1:8000/',
        template: require('grunt-template-jasmine-requirejs'),
        templateOptions: {
          requireConfig: {
            baseUrl: './',
            paths: {
              'jasmine_rsvp': 'path/to/jasmine_rsvp'
            },
            deps: [ 'jasmine_rsvp' ],
            callback: function($) {
            }
          }
        }
      }
    }
  }
});
```

Configure `jasmine_rsvp` in the `paths: { ... }` hash to where you've downloaded the file from `dist/` and you should be good to go.

**Edit**: I'm not sure how well that plays out with tests that are not wrapped in a `require()` or `define()` call, but I assume it shouldn't matter as long as the template does include your test files. If it still doesn't work, please write the output of the `tests.html` file (or whatever the runner generates, see the [keepRunner](https://github.com/gruntjs/grunt-contrib-jasmine#optionskeeprunner) option) either here or in a gist so I can take a better look.

**Comment by ntrrgc** [comment]
> Then it's easier

Personally, that looks very convoluted for me. Is not there any way to get jasmine_rsvp to work without having to load the scripts with XHR and therefore having to do all this stuff? Doesn't require.js have something to build the library in such a way that it does not depend on it?

Anyway, I tried that configuration, but it did not work. It starts the server successfully, it downloads all the Javascript files correctly (checked in Chrome's Network tab) and there are no 404 errors in the console, but the tests fail with lots of reference errors. It looks like it's downloading the files but not running them.

Trying to call the functions defined in the code from the console does not work either, so it does not look like this is because tests are executing before the module source code. Interestingly enough, the test suite is loaded fine even though it does not have `require()` or `define()` calls either.

These are the script tags generated in the test runner, shortened for clarity:

```
<script>
require.config({
  "baseUrl": "./",
  "paths": {
    "jasmine_rsvp": "testlib/jasmine_rsvp-full"
  },
  "deps": [
    "jasmine_rsvp"
  ],
  "callback": function anonymous($) {
    launchTest();
  }
});
</script>
<script>
  // Delay onLoad function until we are ready
  var startTests = window.onload;
  window.onload = null;

  function launchTest() {
  require([
      'lib/my.class',
      'lib/signals',
      'src/snorky',
      'src/services/base',
      'src/services/messaging',
      [...]
  ],
  function(){
    require([
      'src/spec/class.spec.js',
      'src/spec/collections.spec.js',
      [...],
      '.grunt/grunt-contrib-jasmine/reporter.js'
    ], function(){
      startTests();
    });
  })
}
</script>
```

**Comment by amireh** [comment]
> Personally, that looks very convoluted for me.

Welcome to the world of JavaScript module management. Personally, I've stuck with AMD and been using it everywhere, but it may make sense to check out an alternative for these libraries.

> Doesn't require.js have something to build the library in such a way that it does not depend on it?

Not quite.

I'll take a look at this over the week and see how it goes.

**Comment by amireh** [comment]
A year later, here's a gist that shows how to achieve the same effect this library does but without the requirejs crap: https://gist.github.com/amireh/a31d56b8c638be81c3e1
